mlir/test/Dialect/Tosa/tosa-infer-shapes.mlir

```diff
@@ -1310,3 +1310,14 @@ func.func @test_large_constant_permutation() {
   return
 }
 
+// -----
+
+// CHECK-LABEL: test_rank0_transpose_perms
+// Fail to infer the shape but not crash.
+func.func @test_rank0_transpose_perms() {
+  %14 = tensor.empty() : tensor<5x27xi64>
+  %cst = tensor.empty() : tensor<i32>
+  // CHECK: tosa.transpose
+  %72 = tosa.transpose %14, %cst : (tensor<5x27xi64>, tensor<i32>) -> tensor<?x?xi64>
```

Comment:
Based on the description (in the TOSA Spec) of `tosa.transpose` it seems like this op should be illegal. The permutation tensor is expected to be a rank-1 tensor with an element count equal to the rank of the input tensor.

Reply:
I agree, at the moment, `transpose` doesn't have a verifier so this was just a crash-fix PR but I will add a verifier which checks for those thing in a separate PR.